internal/builders/bazel/build.sh

```diff
@@ -25,20 +25,23 @@ IFS=' ' read -r -a BUILD_TARGETS <<< "${TARGETS}"
 # Build with respect to entire arrays of flags and targets
 bazel build "${BUILD_FLAGS[@]}" "${BUILD_TARGETS[@]}"
 
-# Using target string, copy artifact to binaries dir
-for CURR_TARGET in "${BUILD_TARGETS[@]}"; do
-  # Take out the first two // in CURR_TARGET
-  # "//src/internal:fib" --> "src/internal:fib"
-  CD_PATH=$(echo "$CURR_TARGET" | cut -d'/' -f3-)
-
-  # Removes field after and including the colon
-  # "src/internal:fib" --> "src/internal"
-  CD_PATH=$(echo "$CD_PATH" | cut -d':' -f1)
+# Use associative array as a set to increase efficency in avoiding double copying the target
+declare -A FILES_SET
 
-  # Removes everything up to and including the first colon
-  # "//src/internal:fib" --> "fib"
-  BINARY_NAME=${CURR_TARGET#*:}
+# Using target string, copy artifact(s) to binaries dir
+for CURR_TARGET in "${BUILD_TARGETS[@]}"; do
+  # Get file(s) generated from build with respect to the target
+  bazel_generated=$(bazel cquery --output=starlark --starlark:expr="'\n'.join([f.path for f in target.files.to_list()])" "$CURR_TARGET" 2>/dev/null)
+  
+  # Uses a Starlark expression to pass new line seperated list of file(s) into the set of files
+  while read -r file; do
+    # Key value is target path, value we do not care about and is set to constant "1"
+    FILES_SET["${file}"]="1"
+  done <<< "$bazel_generated"
+done
 
-  # Copy the binary to artifact directory, binaries
-  cp "bazel-bin/$CD_PATH/$BINARY_NAME" ./binaries
+# Copy set of unique targets to binaries. Without !, it would give values not keys
+for file in "${!FILES_SET[@]}"; do
+  # Remove the symbolic link and copy
+  cp -L "$file" ./binaries
```

Comment:
Also, consider copying files to upload into a temp directory rather than `./binaries` since this is in the `GITHUB_WORKSPACE` which is typically a checkout of the user's repo so you technically could run into name collisions.

Could be on a later PR.

Reply:
Thanks for the idea! I will implement that later. Tracking it on this [issue](https://github.com/slsa-framework/slsa-github-generator/issues/2331).